components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/RDBMSSessionDataStore.java

```diff
@@ -119,44 +102,29 @@
-    private static boolean tempDataCleanupEnabled = false;
-    private final String sqlInsertSTORE;
-    private final String sqlInsertDELETE;
-    private final String sqlDeleteTempDataTask;
-    private final String sqlDeleteDELETETask;
+    private static int maxSessionDataPoolSize;
+    private static int maxTempDataPoolSize;
+    private static boolean tempDataCleanupEnabled;
     private boolean enablePersist;
-    private String sqlDeleteSTORETask;
     private String sqlSelect;
     private String sqlDeleteExpiredDataTask;
-    private int deleteChunkSize = DEFAULT_DELETE_LIMIT;
-    private boolean sessionDataCleanupEnabled = true;
-    private boolean operationDataCleanupEnabled = false;
+    private int deleteChunkSize;
+    private boolean sessionDataCleanupEnabled;
 
     {
         try {
-            String maxPoolSizeValue = IdentityUtil.getProperty("JDBCPersistenceManager.SessionDataPersist.PoolSize");
-            if (StringUtils.isNotBlank(maxPoolSizeValue)) {
-                if (log.isDebugEnabled()) {
-                    log.debug("Session data pool size config value: " + maxPoolSizeValue);
-                }
-                maxSessionDataPoolSize = Integer.parseInt(maxPoolSizeValue);
-            }
-
-            String isTempDataCleanupEnabledVal = IdentityUtil.getProperty("JDBCPersistenceManager.SessionDataPersist.TempDataCleanup.Enable");
-            if (StringUtils.isNotBlank(isTempDataCleanupEnabledVal)) {
-                tempDataCleanupEnabled = Boolean.parseBoolean(isTempDataCleanupEnabledVal);
+            maxSessionDataPoolSize = getIntProperty(FrameworkConstants.SessionDataStoreConstants.GET_POOL_SIZE,
+                    FrameworkConstants.SessionDataStoreConstants.DEFAULT_MAX_SESSION_DATA_POOLSIZE);
+            if (log.isDebugEnabled()) {
+                log.debug("Session data pool size config value: " + maxSessionDataPoolSize);
             }
-
-            String maxTempDataPoolSizeValue = IdentityUtil.getProperty("JDBCPersistenceManager.SessionDataPersist.TempDataCleanup.PoolSize");
-            if (StringUtils.isNotBlank(maxTempDataPoolSizeValue)) {
-                if (log.isDebugEnabled()) {
-                    log.debug("Temporary data pool size config value: " + maxPoolSizeValue);
-                }
-                maxTempDataPoolSize = Integer.parseInt(maxTempDataPoolSizeValue);
+            tempDataCleanupEnabled = getBooleanProperty(FrameworkConstants.SessionDataStoreConstants.GET_TEMP_CLEANUP_ENABLE,
+                    FrameworkConstants.SessionDataStoreConstants.DEFAULT_TEMP_DATA_CLEANUP_ENABLED);
+            maxTempDataPoolSize = getIntProperty(FrameworkConstants.SessionDataStoreConstants.GET_TEMP_POOL_SIZE,
+                    FrameworkConstants.SessionDataStoreConstants.DEFAULT_MAX_TEMP_DATA_POOLSIZE);
+            if (log.isDebugEnabled()) {
+                log.debug("Temporary data pool size config value: " + maxTempDataPoolSize);
             }
```

Comment:
We may have a private method to encapsulate this part as its being re-used within the class. Ex: `getInternalProperty(<property_name>)`. Within the method we can add the debug log.

Reply:
Please fix other places accordingly.